syntax = "proto3";


option objc_class_prefix = "FD";

package fundamental_data;

service FundamentalData {
    rpc ProcessReport (Report) returns (Empty) {}
    rpc GetAllReports (Report) returns (stream Report) {}
    rpc GetLatestReport (Report) returns (Report) {}

    rpc GetLatestFinancialReportDate (StockRequest) returns (ReportMetaData) {}
    rpc IsUpToDate (StockRequest) returns (UpToDate) {}
}

message Empty {}

message Report {
    string stock = 1;
    string reportType = 2;
    string content = 3;
    string insertDate = 4;
}


message StockRequest {
    string stock = 1;
}
message ReportMetaData {
    string reportDate = 1;
    string insertDate = 2;
}

message UpToDate {
    bool upToDate = 1;
    // add other information later on
}